presto-tests/src/main/java/io/prestosql/tests/LogTestExecutionListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests;
+
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import org.testng.IClassListener;
+import org.testng.ITestClass;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LogTestExecutionListener
+        implements IClassListener
+{
+    private static final Logger LOG = Logger.get(LogTestExecutionListener.class);
+
+    private final Map<String, Long> started = new HashMap<>();
+
+    @Override
+    public void onBeforeClass(ITestClass testClass)
+    {
+        String name = testClass.getName();
+        LOG.info("Running tests from %s ...", name);
+        started.put(name, System.nanoTime());
```

Comment:
This probably works for everything except `io.prestosql.operator.scalar.TestRegexpFunctions`, since the class uses `@Factory`. Please add a TODO for documentation purposes.